course5/homework06/test.h

```diff
@@ -0,0 +1,103 @@
+#include <cstring>
+
+template <typename T>
+class TestClass
+{
+public:
+    void test_const_char_constructor()
+    {
+        auto str = "this is string with length 29";
+        auto s = T(str);
+        assert(s == "this is string with length 29");
+    }
+
+    void test_move_constructor()
+    {
+        auto str = "this is string with length 29";
+        auto s = T(str);
+        auto s2 = std::move(s);
+        assert (s2 == "this is string with length 29");
+    }
+
+    void test_copy_assignment()
+    {
+        auto s1 = T("blaah");
+        T s2 = s1;
+
+        assert (s2.length() == 5);
+        assert (s1.length() == 5);
+        assert (s1 == s2);
```

Comment:
test that the copy is deep, ie that changing s1 after the copy doesn't affect s2